### PR TITLE
[mosquitto] port to common

### PR DESCRIPTION
**Pull request by ishioni**
**Description of the change**
It's on common!

**Benefits**
Common?

**Possible drawbacks**

**Applicable issues**

- fixes #

**Additional information**
non-anonymous access to be implemented once i get enough time to test it out

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Title of the PR starts with chart name (e.g. `[home-assistant]`)
- [X] (optional) Chart is using our [common](https://github.com/k8s-at-home/charts/tree/master/charts/common) library as a dependency.
- [X] (optional) Variables are documented in the README.md

### CODE REVIEW

File: charts/mosquitto/README_CHANGELOG.md.gotmpl [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+### [1.0.0]
+
+#### Added
+
+- Initial release
+
+#### Changed
+
+- N/A
```
Comment (bjw-s) [inline charts/mosquitto/README_CHANGELOG.md.gotmpl:20]:
```suggestion
#### Added

- N/A

#### Changed

- Migrated the chart to the [k8s-at-home common library chart](https://github.com/k8s-at-home/charts/tree/master/charts/common).
```

File: charts/mosquitto/templates/common.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{{/* Make sure all variables are set properly */}}
+{{- include "common.values.setup" . }}
+
+{{/* Append the configMap to the additionalVolumes */}}
+{{- define "mosquitto.configmap.volume" -}}
+name: mosquitto-settings
```
Comment (bjw-s) [inline charts/mosquitto/templates/common.yaml:6]:
```suggestion
name: mosquitto-config
```

File: charts/mosquitto/templates/common.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+{{- $volume := include "mosquitto.configmap.volume" . | fromYaml -}}
+{{- if $volume -}}
+    {{- $additionalVolumes := append .Values.additionalVolumes $volume }}
+    {{- $_ := set .Values "additionalVolumes" (deepCopy $additionalVolumes) -}}
+{{- end -}}
+
+{{/* Append the configMap volume to the additionalVolumeMounts */}}
+{{- define "mosquitto.configmap.volumeMount" -}}
+name: mosquitto-settings
```
Comment (bjw-s) [inline charts/mosquitto/templates/common.yaml:19]:
```suggestion
name: mosquitto-config
```

**Review by bjw-s — CHANGES_REQUESTED** [review]
Looks good! 👍 Just a few cosmetic changes requested 🙂

File: charts/mosquitto/templates/common.yaml (hunk 3 above)
Comment (ishioni) [reply charts/mosquitto/templates/common.yaml:19]: I've gotten away with -settings on transmission ;) But no problem

Comment (bjw-s) [reply charts/mosquitto/templates/common.yaml]: 😂 I just had my morning coffee, I'm probably more alert now ;)

**Review by bjw-s — APPROVED** [review]

**Pull request merged**